### ISSUE TITLE
nomad status - 404 not found

### ISSUE THREAD

**Issue by BSick7** [hashicorp/nomad#1650]
If filing a bug please include the following:
### Nomad version

clients and servers

```
$ nomad -v
Nomad v0.4.1
```
### Issue

If a job placement fails, `nomad status <job>` will say job is not found.

I would expect the "failed placement" message from `eval-status` to show in this status.
### Reproduction steps
1. Issue a job with a constraint that cannot pass
2. Issue `nomad status <job-id>`
### Nomad Client logs (if appropriate)

```
$ nomad status cpm
ID          = cpm
Name        = cpm
Type        = service
Priority    = 50
Datacenters = us-east-1a,us-east-1b,us-east-1c
Status      = running
Periodic    = false
Error querying job summary: Unexpected response code: 404 (job not found)
```

```
$ nomad eval-status 8f9f7153-90fd-ea46-3774-a8088f97d83f
ID                 = 8f9f7153
Status             = complete
Status Description = complete
Type               = service
TriggeredBy        = job-register
Job ID             = cpm
Priority           = 50
Placement Failures = true

Failed Placements
Task Group "deploy-green" (failed to place 1 allocation):
  * Class "blue" filtered 3 nodes
  * Constraint "${node.class} = green" filtered 3 nodes

Evaluation "d6e8c64d" waiting for additional capacity to place remainder
```

**Comment by schmichael** [comment]
Hm, I've been unable to reproduce this by running a job with a `windows` constraint on a single `linux` host. I tried putting the constraint on both the job and the task.

* Config files: https://gist.github.com/schmichael/ee9860626bfad252a29c18e393f43669

In 4 different terminals I ran:

```sh
consul agent -dev -advertise=127.0.0.1 # so client can discover server
nomad-0.4.1 agent -config server.hcl
nomad-0.4.1 agent -config client1.hcl

# In the fourth terminal run:
nomad-0.4.1 run example.hcl
nomad-0.4.1 status example

# Output:
ID          = example
Name        = example
Type        = service
Priority    = 50
Datacenters = dc1
Status      = pending
Periodic    = false

Summary
Task Group  Queued  Starting  Running  Failed  Complete  Lost
cache       1       0         0        0       0         0

Placement Failure
Task Group "cache":
  * Constraint "${attr.kernel.name} = windows" filtered 1 nodes

Allocations
No allocations placed
```

(I have multiple nomad versions installed, hence the version number suffix.)

@BSick7 Can you share a sample job file to reproduce this?

**Comment by BSick7** [comment]
Does your evaluation report this?

> Evaluation "d6e8c64d" waiting for additional capacity to place remainder

**Comment by schmichael** [comment]
Were all of your nodes on 0.4.1?

**Comment by BSick7** [comment]
Yes, they were. Sorry for reporting so late.

**Comment by hgontijo** [comment]
I just faced the same 404 (job not found) issue.

```
$ nomad -v
Nomad v0.4.1

$ nomad status CC-Continuous-177_ABV3Server
ID          = CC-Continuous-177_ABV3Server
Name        = CC-Continuous-177_ABV3Server
Type        = batch
Priority    = 50
Datacenters = us-west-2
Status      = running
Periodic    = false
Error querying job summary: Unexpected response code: 404 (job not found)

$ nomad eval-status -verbose eeb731d5
ID                 = eeb731d5-daac-7c81-5f06-fa28eb77e49c
Status             = complete
Status Description = complete
Type               = batch
TriggeredBy        = job-register
Job ID             = CC-Continuous-177_ABV3Server
Priority           = 50
Placement Failures = false
Previous Eval      = <none>
Next Eval          = <none>
Blocked Eval       = <none>
```

All logs related to this jobs are available [here](https://gist.github.com/hgontijo/f5e08be87a26af3ddf49a7a5f1c76b8d).

**Comment by dadgar** [comment]
@hgontijo Thanks for the logs. Can you add some color onto what was happening with that job. Was it getting run and stopped multiple times? A bit of timeline would be helpful

**Comment by hgontijo** [comment]
Here's what's happening for me. The macOS Nomad binary is making an additional call `GET /v1/job/<job_name>/summary` and gets a `HTTP/1.1 404 Not Found`  response.
It works fine on Linux.

# Linux 
```
$ uname -a
Linux ip-10-60-1-60 4.4.23-31.54.amzn1.x86_64 #1 SMP Tue Oct 18 22:02:09 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

$ nomad version
Nomad v0.4.0
## Tested against Nomad server 0.4.0.

$ strace -s 500 nomad status -address=http://internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646 CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9
...
write(3, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9 HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n", 239) = 239
write(3, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9/allocations HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n", 251) = 251
write(3, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9/evaluations HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n", 251) = 251
...
```

# macOS
```
$ uname -a
Darwin hgontijo-mbp15.foo.com 15.6.0 Darwin Kernel Version 15.6.0: Thu Sep  1 15:01:16 PDT 2016; root:xnu-3248.60.11~2/RELEASE_X86_64 x86_64 i386 MacBookPro11,2 Darwin

$ nomad version
Nomad v0.4.0
## I also tested with Nomad client 0.4.1, 0.5.0. against Nomad server 0.4.0.

$ dtruss nomad status -address=http://internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646 CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9
write(0x6, "GET /v1/jobs?prefix=CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9 HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n\0", 0xF7)    = 247 0
write(0x6, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9 HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n\0", 0xEF)    = 239 0
write(0x6, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9/allocations HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n\0", 0xFB)    = 251 0
write(0x6, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9/evaluations HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n\0", 0xFB)    = 251 0
write(0x6, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9/summary HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n\0", 0xF7)                = 247 0
select(0x0, 0x0, 0x0, 0x0, 0x700000080DC8)               = 0 0
kevent(0x7, 0x0, 0x0)            = 1 0
kevent(0x7, 0x0, 0x0)            = 1 0
read(0x6, "HTTP/1.1 404 Not Found\r\nContent-Encoding: gzip\r\nContent-Type: application/x-gzip\r\nDate: Thu, 08 Dec 2016 23:27:02 GMT\r\nVary: Accept-Encoding\r\nX-Nomad-Index: 19880\r\nX-Nomad-Knownleader: true\r\nX-Nomad-Lastcontact: 0\r\nContent-Length: 37\r\nConnection: Close\r\n\r\n", 0x1000)              = 293 0
```

**Comment by dadgar** [comment]
@hgontijo Are you able to reproduce this reliably? If so could you provide steps? I tried locally and on a cluster and couldn't.

I submitted jobs with constraints that didn't match and then would run `nomad status` on them and never hit the 404.

**Comment by hgontijo** [comment]
@dadgar yes, I'm able to reproduce this reliably. The error is on a call (GET /v1/job/\<id>/summary) to an endpoint that does not exist:
`
write(0x6, "GET /v1/job/CC-Nightly_Tests-3_CCNewSmokeDB-Oracle-9/summary HTTP/1.1\r\nHost: internal-tf-lb-20161207190125756957107qlo-99215739.us-west-2.elb.amazonaws.com:4646\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\nConnection: close\r\n\r\n\0", 0xF7)                = 247 0
`

Nomad server and client are v0.4.0.
It seems [/v1/job/\<id>/summary](https://github.com/hashicorp/nomad/commit/38b1ba7e4e9bbc1ae9d80aa139abf124573aecef)  endpoint was added v0.4.1 so I still don't understand why it's failing on v0.4.0.

**Comment by dadgar** [comment]
@hgontijo I see, you are on an older version of Nomad. Is the `nomad` binary you are using to do things like `nomad status` a newer version than the servers/clients?

**Comment by hgontijo** [comment]
@dadgar I executed those `nomad status` commands from my machine with Nomad versions 0.4.0, 0.4.1 and 0.5.0, all of them fails. The Nomad servers/clients are 0.4.0.

**Comment by dadgar** [comment]
Yeah I would suggest using the same version as the servers/clients. Can you try again on 0.4.0, that binary never calls any summary API as it didn't exist at the time.

https://github.com/hashicorp/nomad/blob/v0.4.0/command/status.go
versus:
https://github.com/hashicorp/nomad/blob/v0.4.1/command/status.go#L233

**Comment by hgontijo** [comment]
I found out the mismatch issue on my machine with 0.4.0.
I just download again [0.4.0](https://releases.hashicorp.com/nomad/0.4.0/nomad_0.4.0_darwin_amd64.zip) binary and it's working fine. :) 

Previously I was running a 0.4.0 that I downloaded and compiled from [archive] (https://github.com/hashicorp/nomad/archive/v0.4.0.tar.gz) and that was failing. Wondering what made that 0.4.0 compiled on my machine to have such weird behavior. I still have the binary if you're interested to take a look.
